### PR TITLE
osconfig_agent: add inventory tests, fix writeInventory function

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: cli_tools/osconfig_agent/main.go [hunk 1]
```diff
@@ -140,8 +140,8 @@ func run(ctx context.Context) {
 		} else {
 			setOsConfig(resp)
 			setPatchPolicies(resp.PatchPolicies)
-			runInventory()
 		}
+		runInventory()
```
Comment (wj-chen) [inline cli_tools/osconfig_agent/main.go]: Curious about the reason behind this change

File: cli_tools/osconfig_agent/inventory_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	var buf bytes.Buffer
+	io.Copy(&buf, zr)
+	zr.Close()
+
+	var pkgs packages.Packages
+	json.Unmarshal(buf.Bytes(), &pkgs)
+	return pkgs
+}
+
+func TestWriteInventory(t *testing.T) {
```
Comment (wj-chen) [inline cli_tools/osconfig_agent/inventory_test.go]: Do we have the ability to test an unhappy case (check the Error []string field) yet?

File: cli_tools/osconfig_agent/main.go (hunk 1 above)
Comment (adjackura) [reply cli_tools/osconfig_agent/main.go]: The inventory run does not require a non error from the osconfig api. This is a bit of a hack to get around projects that don't have osconfig enabled but want to use inventory data. We will have a better solution in the future.

File: cli_tools/osconfig_agent/inventory_test.go (hunk 2 above)
Comment (adjackura) [reply cli_tools/osconfig_agent/inventory_test.go]: The Error field is going to be removed in favor of reporting errors via some other method, and then we will have to add a test for the various failure modes

**Pull request merged**